### ISSUE TITLE
A question in my custom loss

### ISSUE THREAD

**Issue by ft512835** [keras-team/keras#8993]
Hi,  I'm going to write an custom loss, which based on 'categorical_crossentropy'.
The versions:
Keras 2.1.2
Tensorflow 1.4

e.g. 
`def custom_loss(y_true, y_pred, e=0.1):`
`return (1-e)*K.categorical_crossentropy(y_pred, y_true) +
e*K.categorical_crossentropy(y_pred, K.ones_like(y_pred)/NUM_CLASSES)`

and complie like this,
`model.compile(optimizer=opt, loss=custom_loss, metrics=['acc'])`

Then train, I found the loss is much higher than which complie with "categorical_crossentropy" directly, 
`model.compile(optimizer=opt, loss="categorical_crossentropy", metrics=['acc'])`
cannot get expected results.

But this custom_loss works quite well at Keras 2.0.6, Tensorflow 1.2.
What's wrong with this?

**Comment by FrancisYizhang** [comment]
import keras.backend as K

def mean_pred(y_true, y_pred):
    return K.mean(y_pred)

model.compile(optimizer='rmsprop',
              loss='binary_crossentropy',
              metrics=['accuracy', mean_pred])

**Comment by FrancisYizhang** [comment]
@ft512835

**Comment by ft512835** [comment]
I just mess up the order of 'y_pred, y_true'.  It works when I change to K.categorical_crossentropy(y_true, y_pred). 
 @FrancisYizhang  what problem with your loss function?
